### PR TITLE
Remove training mode assertion in `get_text_pe()`

### PR DESCRIPTION
**Pull request by Y-T-G**
https://community.ultralytics.com/t/how-to-run-object-detection-inference-with-a-yoloe-segmentation-model/1436/5

Doesn't seem to affect the output whether it's in training mode or not.

```
from ultralytics import YOLO
model = YOLO("yoloe-11l.yaml").load("yoloe-11l-seg.pt")
names = ['person']
model.set_classes(names, model.get_text_pe(names))
```

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Removes a training-mode restriction so text positional embeddings can be retrieved during training, improving flexibility for multimodal workflows. ✅

### 📊 Key Changes
- Deleted a guard (`assert not self.training`) in `get_text_pe` within `ultralytics/nn/tasks.py`.
- Now allows `get_text_pe` to be called while the model is in training mode.
- No API signature changes; behavior is more permissive and consistent across modes.

### 🎯 Purpose & Impact
- Enables end-to-end training scenarios that need text features or text positional embeddings mid-training (e.g., multimodal/text-conditioned models). 🤖📝
- Prevents crashes/assertion errors when calling `get_text_pe` inside training loops, improving developer experience. 🚀
- Potential consideration: calling this during training may introduce gradient flow and extra compute/memory; users should ensure this is intended in their pipeline. 🧠⚙️

Example (now valid during training):
```python
model.train()
txt_pe = model.get_text_pe(text=["cat", "dog"])  # previously asserted; now allowed
```

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Y-T-G, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! This is an automated response to help speed things up—an Ultralytics engineer will review and assist soon. In the meantime, please review the checklist below and a few PR-specific requests to ensure a smooth merge:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

PR-specific notes and requests:
- 🧩 Scope clarification: Since this removes `assert not self.training` in `get_text_pe()` (`ultralytics/nn/tasks.py`), please confirm intended behavior for training vs. inference and whether any models in the repo rely on the previous assertion behavior.
- 🧪 Tests: Add or update unit tests that exercise `get_text_pe()` in both training and eval modes, including the `YOLOEDetect` head path (`head.get_tpe`) to ensure no regressions.
- 🔍 Safety checks: Please verify compatibility with common workflows (e.g., DDP, AMP/mixed precision, export, and TorchScript if applicable) and note any observed changes in compute/memory during training.
- 📝 Docs: If this changes expected usage or developer assumptions for text-aware heads, add a short note to the relevant [documentation](https://docs.ultralytics.com/).
- 🔗 Traceability: If this change addresses a specific problem discussed in issues, link the corresponding [GitHub issue](https://github.com/ultralytics/ultralytics/issues) in the PR description for tracking.
- 🧹 Code style: Ensure the change is minimal and localized to the necessary lines without affecting unrelated logic.

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don’t hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀✨

**Review by Laughing-q — APPROVED** [review]

**Comment by UltralyticsAssistant** [comment]
Absolutely thrilled to see this merged! Thanks @Y-T-G for the thoughtful update in PR [#22076](https://github.com/ultralytics/ultralytics/pull/22076), and kudos to @glenn-jocher and @Laughing-q for the collaboration. “The best way to predict the future is to invent it.” —Alan Kay. By lifting the training-mode restriction on `get_text_pe`, you’ve opened the door for smoother multimodal workflows and more flexible end-to-end training. This is a small change with big impact—appreciate the precision and the care for developer experience. 🚀
